lib/auth/auth_with_roles.go

```diff
@@ -4212,6 +4212,15 @@ func (a *ServerWithRoles) GenerateSnowflakeJWT(ctx context.Context, req *proto.S
 	return a.authServer.GenerateSnowflakeJWT(ctx, req)
 }
 
+// CreateSAMLIdPResponse creates a SAML response to a SAML auth request.
+func (a *ServerWithRoles) CreateSAMLIdPResponse(ctx context.Context, req *proto.CreateSAMLIdPResponseRequest) (*proto.CreateSAMLIdPResponseResponse, error) {
+	// Only the proxy can create SAML IdP responses.
+	if !a.hasBuiltinRole(types.RoleDatabase, types.RoleProxy) {
```

Comment:
guessing either RoleDatabase shouldn't be here or the comments/errors should be updated
```suggestion
	if !a.hasBuiltinRole(types.RoleProxy) {
```

Reply:
i think you already fixed this actually